test/e2e_tests/utils/nfspvc_adapter.go

```diff
@@ -40,6 +40,7 @@ func CreateNfsPvc(k8sClient client.Client, nfsPvc *nfspvcv1alpha1.NfsPvc) *nfspv
 	nfsPvcName := nfsPvc.Name + "-" + randString
 	newNfsPvc := nfsPvc.DeepCopy()
 	newNfsPvc.Name = nfsPvcName
+	newNfsPvc.Spec.NfsVersion = "v4"
```

Comment:
use the const instead of hardcoding.
In general, I would rather you remove this from the generic adapter and only add this field in the tests where it is relevant.